File: app/src/main/java/com/akrivonos/a2chparser/adapters/viewholders/BoardConcreteViewHolder.java
package com.akrivonos.a2chparser.adapters.viewholders;

import android.view.View;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.akrivonos.a2chparser.R;
import com.akrivonos.a2chparser.database.RoomAppDatabase;
import com.akrivonos.a2chparser.interfaces.OpenBoardListener;
import com.akrivonos.a2chparser.models.database.Board;
import com.akrivonos.a2chparser.pojomodel.boardmodel.BoardConcrete;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;

public class BoardConcreteViewHolder extends RecyclerView.ViewHolder {

    public final TextView nameBoardTextView;
    public final TextView idBoardTextView;
    public final ImageButton imageButtonSave;
    private Board board;
    private RoomAppDatabase database;

    public BoardConcreteViewHolder(@NonNull View itemView, final OpenBoardListener openBoardListener, RoomAppDatabase database) {
        super(itemView);
        nameBoardTextView = itemView.findViewById(R.id.name_board);
        idBoardTextView = itemView.findViewById(R.id.id_board);
        imageButtonSave = itemView.findViewById(R.id.button_is_save);
        this.database = database;

        itemView.setOnClickListener(view -> {
            if (board != null) {
                openBoardListener.openBoard(board);
            }
        });
    }

    public Disposable setBoardConcrete(BoardConcrete boardConcrete) {
        this.board = new Board(boardConcrete);
        nameBoardTextView.setText(boardConcrete.getName());
        idBoardTextView.setText("/".concat(boardConcrete.getId()).concat("/"));
        return database.boardsDao().getBoardById(boardConcrete.getId())
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(list -> imageButtonSave.setBackgroundResource((list.size() != 0)
                        ? R.drawable.ic_star_save
                        : R.drawable.ic_star_not_save));
    }
}

File: app/src/main/java/com/akrivonos/a2chparser/adapters/viewholders/BoardConcreteViewHolder.kt
package com.akrivonos.a2chparser.adapters.viewholders

import android.util.Log
import android.widget.ImageButton
import androidx.recyclerview.widget.RecyclerView
import com.akrivonos.a2chparser.R
import com.akrivonos.a2chparser.database.RoomAppDatabase
import com.akrivonos.a2chparser.databinding.AdapteritemConcreteBoardBinding
import com.akrivonos.a2chparser.interfaces.OpenBoardListener
import com.akrivonos.a2chparser.models.database.Board
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers

class BoardConcreteViewHolder(var binding: AdapteritemConcreteBoardBinding, openBoardListener: OpenBoardListener, private val database: RoomAppDatabase) : RecyclerView.ViewHolder(binding.root) {

    private val imageButtonSave: ImageButton = itemView.findViewById(R.id.button_is_save)

    init {
        itemView.setOnClickListener {
            openBoardListener.openBoard(binding.board)
        }
        imageButtonSave.setOnClickListener {
            (if (binding.isSaveState)
                database.boardsDao().deleteBoard(binding.board?.idBoard)
            else
                database.boardsDao().saveBoard(binding.board)
                    )
                    .subscribeOn(Schedulers.io())
                    .observeOn(Schedulers.io())
                    .subscribe()
        }
    }

    fun bind(board: Board): Disposable {
        binding.board = board
        return database.boardsDao().getBoardById(board.idBoard)
                .subscribeOn(Schedulers.io())
                .map { list -> list.isNotEmpty() }
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe { isSaveLocal ->
                    Log.d("test", "isSaveLocal: $isSaveLocal id: ${board.idBoard}")
                    binding.isSaveState = isSaveLocal
                    binding.executePendingBindings()
                }
    }
}
